File: rpl/Exercises/B - Division y Conquista/10 - Elemento desordenado/main.py
l = [1, 2, 90, 5, 6, 7]

def elemento_desordenado_dyc(l, start, end):
    if (end - start == 1):
        return start if l[start] > l[end] else None
    
    mid = (start + end) // 2
  
    return elemento_desordenado_dyc(l, start, mid) or elemento_desordenado_dyc(l, mid, end)


print(l[elemento_desordenado_dyc(l, 0, len(l)-1)])

File: rpl/Exercises/B - Division y Conquista/10 - Elemento desordenado/main.c
#include "main.h" // No borrar

int elemento_desordenado_dyc(int arr[], size_t inicio, size_t fin) {
    if (fin - inicio == 1)
        return arr[inicio] > arr[fin] ? (int)inicio : NULL;

    size_t medio = (inicio + fin) / 2; 

    int izq = elemento_desordenado_dyc(arr, inicio, medio);

    return izq ? izq : elemento_desordenado_dyc(arr, medio, fin);
}

int elemento_desordenado(int arr[], size_t n) {
    return arr[elemento_desordenado_dyc(arr, 0, n-1)];
}